dbms/src/Storages/Page/V3/Universal/S3PageReader.cpp

```diff
@@ -12,34 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <IO/ReadBufferFromFile.h>
+#include <IO/ReadBufferFromRandomAccessFile.h>
 #include <Storages/Page/V3/Universal/S3PageReader.h>
 #include <Storages/Page/V3/Universal/UniversalPageIdFormatImpl.h>
 #include <Storages/S3/S3Common.h>
+#include <Storages/S3/S3Filename.h>
 #include <Storages/S3/S3RandomAccessFile.h>
 
 namespace DB::PS::V3
 {
+S3PageReader::S3PageReader()
+    : s3_client(S3::ClientFactory::instance().sharedClient())
+    , bucket(S3::ClientFactory::instance().bucket())
```

Comment:
removed. only get them from the factory when we need send request to S3

Reply:
Fixed